code/modules/events/vox_trade_probe.dm

```diff
@@ -0,0 +1,116 @@
+/*
+ * Vox Trade Probe - if no trader, trade shuttle in dock, probe has not fired yet, wages are not below 100%, port is open
+ * Fire weight can increase up to 90, which is very high for an event, if lot of players, money, and time
+ */
+
+ #define TRADEPROBE_BONUS_TIMER 120 //how many seconds need to pass to increase weight by 1
+
+/datum/event/tradeprobe
+	endWhen	= 270 //How many 2-second ticks should we stay? 9 minutes = 540 / 2 = 270
+	oneShot = TRUE //one per shift
+
+/datum/event/tradeprobe/can_start(var/list/active_with_role)
+	//No probe if... there is a trader, the trade shuttle is at the station, or financial mismanagement
+	if(SStrade.loyal_customers.len)
+		return FALSE
+	if(!istype(trade_shuttle.current_port,/obj/docking_port/destination/trade/start))
+		return FALSE
+	if(!ports_open)
+		command_alert(/datum/command_alert/tradeaversion_closedport)
+		return FALSE
+	if(payroll_modifier < 1)
+		command_alert(/datum/command_alert/tradeaversion_mismanagement)
+		return FALSE
+
+	//Gain 1 weight per 3% above normal wages players received last cycle, up to 30 weight at 190%
+	var/wageboost = min((payroll_modifier-1)/0.03,30)
+	//Gain 5 weight per player beyond 10, up to 30 weight at 20 players
+	var/playerboost = min(max(0,active_with_role["Any"]-10) * 3, 30)
+	//Gain 1 weight per 2 minutes after shieft start, up to 30 weight at 60 minutes
+	var/timeboost = min(((world.time / (1 SECONDS)) - ticker.gamestart_time)/(TRADEPROBE_BONUS_TIMER),30) //gamestart_time is in seconds
+	return wageboost + playerboost + timeboost
+
+/datum/event/tradeprobe/setup()
+	//Lock trade shuttle so that it can't crush the probe
+	trade_shuttle.lockdown = "The trade shuttle has been temporarily locked while a pod is in port. The pod will depart after 10 minutes."
+	load_dungeon(/datum/map_element/dungeon/tradeprobe)
+	tradeprobe = new(starting_area = /area/shuttle/tradeprobe)
+	tradeprobe.initialize()
+	tradeprobe.move()
+	/*tradeprobe.set_transit_dock(/obj/docking_port/destination/trade/station)
+	tradeprobe.move_to_dock(tradeprobe.destination_port)*/
+
+/datum/event/tradeprobe/announce()
+	command_alert(/datum/command_alert/tradeprobe)
+
+/datum/event/tradeprobe/end()
+	command_alert(/datum/command_alert/tradeprobe_depart)
+	spawn(1 MINUTES)
+		trade_shuttle.lockdown = FALSE
+		//Send shuttle to secondary dock at trade outpost
+		tradeprobe.remove_dock(/obj/docking_port/destination/trade/station)
+		tradeprobe.add_dock(/obj/docking_port/destination/salvage/trading_post)
+		tradeprobe.destroy_everything = TRUE //One last trip
+		tradeprobe.move()
+		for(var/obj/structure/trade_window/TW in SStrade.all_twindows)
+			TW.new_pending(tw_probe_return)
+
+/obj/machinery/vending/sale/trader/probe
+	anchored = TRUE
+	mech_flags = MECH_SCAN_FAIL
+
+/obj/machinery/vending/sale/trader/probe/New()
+	..()
+	//Grab some items from variety packs...
+	for(var/i = 1 to 6)
```

Comment:
`6` should be a define so that a new coder can easily change how many objects spawn, in my opinion.